### ISSUE TITLE
Study: make `USE_PSA_CRYPTO` compatible with `ECP_RESTARTABLE`

### ISSUE THREAD

**Issue by mpg** [ARMmbed/mbedtls#5400]
### Suggested enhancement

Currently the two compile-time options `MBEDTLS_USE_PSA_CRYPTO` and `MBEDTLS_ECP_RESTARTABLE` are mutually incompatible, because the first one wants to use `psa_` APIs which currently don't support restartable operations.

Study how to make those options compatible with the following meaning: when an operation needs to be restartable (because that was enabled not only at build-time, but also at runtime for this operation), use the `mbedtls_` API (which is the only option for now), otherwise use the `psa_` API. For example, in `mbedtls_pk_sign_restartable()` and `mbedtls_pk_verify_restartable()`, if `rs_ctx == NULL` or if `mbedtls_ecp_set_max_ops()` was not called, use the `psa_` API, otherwise use the `mbedtls_` API. Similar decisions will need to be made for ECDH in the TLS code.

Task outcome: a list of estimated tasks achieving the desired result.

### Justification

Mbed TLS needs this because we want to eventually make `MBEDTLS_USE_PSA_CRYPTO` active by default (see #5156) in order to avoid maintaining two versions of the code.

**Comment by gilles-peskine-arm** [comment]
Note that eventually the PSA API will support restartable operations, so this task may not be needed.

**Comment by mpg** [comment]
Indeed, and our time would be better spent working on https://github.com/orgs/Mbed-TLS/projects/1#column-18816849 than on this issue. I'm closing it for now, we can always re-open if support for interruptible operations in PSA doesn't progress quickly enough and there is strong demand for this in the meantime.
